http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -1165,14 +1279,29 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
           )
           .transaction
 
-        dbName <- createDatabaseInWsm(app,
+        dbNamePrefixes = app.appType match {
+          case AppType.Wds          => List("wds")
+          case AppType.WorkflowsApp => List("cbas", "cromwellmetadata")
+          case _                    => List()
+        }
+
+        dbNames <-
+          dbNamePrefixes
+            .traverse(databaseNamePrefix =>
+              for {
+                databaseName <-
+                  createDatabaseInWsm(app,
                                       workspaceId,
                                       namespace,
-                                      "wds",
+                                      databaseNamePrefix,
                                       wsmApi,
                                       Option(createIdentityResponse.getResourceId)
-        )
-      } yield (Some(ServiceAccountName(identityName)), Some(dbName))
+                  )
+              } yield databaseNamePrefix -> databaseName
```

Comment:
My personal heuristic is that a`for` inside another `for` is often a sign that you might want to extract a sub-function